### PR TITLE
bpo-32410: Make SendfileNotAvailableError public

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue32410

### CODE REVIEW

File: Lib/asyncio/base_events.py [hunk 1]
```diff
@@ -659,17 +655,17 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
...
             return await self._sock_sendfile_native(sock, file,
                                                     offset, count)
-        except _SendfileNotAvailable as exc:
+        except events.SendfileNotAvailable as exc:
             if fallback:
                 return await self._sock_sendfile_fallback(sock, file,
                                                           offset, count)
             else:
-                raise RuntimeError(exc.args[0]) from None
+                raise
```
Comment (1st1) [inline Lib/asyncio/base_events.py]: Can we reformat this code to:

```python
except events.SendfileNotAvailable:
    if not fallback: 
       raise
return await self._sock_sendfile_fallback(sock, file, offset, count)
```

File: Doc/library/asyncio-eventloop.rst [hunk 2]
```diff
@@ -980,6 +980,18 @@ Handle
...
 
+SendfileNotAvailable
+--------------------
+
+
+.. exception:: SendfileNotAvailable
+
+   Sendfile syscall is not available, subclass of :exc:`RuntimeError`.
+
+   Raised if Operation System does not support senfile syscall for
```
Comment (1st1) [inline Doc/library/asyncio-eventloop.rst]: ... Raised if the OS...

File: Lib/asyncio/base_events.py (hunk 1 above)
Comment (asvetlov) [reply Lib/asyncio/base_events.py]: Done

File: Doc/library/asyncio-eventloop.rst (hunk 2 above)
Comment (asvetlov) [reply Doc/library/asyncio-eventloop.rst]: Done

File: Doc/library/asyncio-eventloop.rst [hunk 3]
```diff
@@ -980,6 +980,18 @@ Handle
       .. versionadded:: 3.7
 
 
+SendfileNotAvailable
+--------------------
+
+
+.. exception:: SendfileNotAvailable
```
Comment (1st1) [inline Doc/library/asyncio-eventloop.rst]: One more nit: can we call it `SendfileNotAvailableError`? The `Error` suffix is used for 99% of exceptions in Python.

Comment (asvetlov) [reply Doc/library/asyncio-eventloop.rst]: Good catch. Done

**Review by 1st1 — APPROVED** [review]

**Comment by orsenthil** [comment]
@asvetlov  - just curious, why did you skip-news for this new Exception introduction?

**Comment by 1st1** [comment]
This is a follow-up PR; we already have a NEWS entry for loop.sock_sendfile.

**Pull request merged**